### PR TITLE
Added a check to ensure memory tools is working

### LINKED ISSUE #14: add function that ensures that memory tools is enabled and working correctly
Due to issues with optimizers, writing a function that actually calls malloc (or one of the other functions) isn't always trivial. memory tools should provide a function (or functions) that ensure the interpose worked (malloc and friends were overridden) and that calling malloc (or other functions) actually gets intercepted.

### PR DESCRIPTION
**Pull request by wjwwood**
Fixes #14

### CODE REVIEW

**Comment by wjwwood** [comment]
CI:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/5287/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/2041/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/4403/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/5266/)

File: osrf_testing_tools_cpp/include/osrf_testing_tools_cpp/memory_tools/register_hooks.hpp [hunk 1]
```diff
@@ -65,6 +70,11 @@ OSRF_TESTING_TOOLS_CPP_MEMORY_TOOLS_PUBLIC
 void
 on_realloc(AnyMemoryToolsCallback callback);
 
+/// Get the current on_realloc callback if set, otherwise null.
+OSRF_TESTING_TOOLS_CPP_MEMORY_TOOLS_PUBLIC
+AnyMemoryToolsCallback
+get_on_realloc();
```
Comment (clalancette) [inline osrf_testing_tools_cpp/include/osrf_testing_tools_cpp/memory_tools/register_hooks.hpp]: Given that we don't actually use `get_on_realloc`/`get_on_calloc`/`get_on_free`, should we bother exposing them?  I generally like to only expose the things that I'm using, to keep the API smaller.

**Comment by wjwwood** [comment]
Thanks for the review, I think I addressed the feedback, can you please re-review when you get some time?

**Comment by wjwwood** [comment]
New CI:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/5331/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/2067/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/4430/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/5298/)

**Pull request merged**